src/horizon_api.ts

```diff
@@ -0,0 +1,317 @@
+import { MemoType, AssetType } from 'stellar-base';
+
```

Comment:
I think if a file's just going to contain type definitions, it should be in a `types` folder and have the `d.ts` extension. This is how we did that in another repo, which I like:

https://github.com/stellar/js-stellar-wallets/tree/master/src/types

I'm not married to having an index file that imports the other type files, but I find it convenient to not have to know what exact file the types are in.

Reply:
The `d.ts` files are picked up by Typescript unconditionally and globally, thus pollutes the scope for Typescript to keep track of, and pollutes auto-complete in IDEs. It would be best to evade `d.ts` and keep in `.ts` files others imports from.

I think that it is a important, machine-readable API definition for Horizon. At least I'm using those types also separately from sdk myself. In my opinion, a new type-only repository would suit best, although that's quite a overkill.